pyqtorch/utils.py

```diff
@@ -153,6 +153,10 @@ def batch_last(operator: Tensor) -> Tensor:
     return torch.permute(operator, undo_perm)
 
 
+class DensityMatrix(Tensor):
```

Comment:
@EthanObadia whats missing here?

Reply:
Nothing. The `DensityMatrix` class is used as a "flag" to ensure the condition in the different `forward` functions. :
```python3
if not isinstance(state, DensityMatrix):
```
It allows to check if the input state is a vector state or a density matrix without any condition on the `size`, `n_qubits`... So `if not isinstance(state, DensityMatrix):` just check a `bool` right?

Reply:
does this work?

Reply:
Yes